File: pdu/python/hako_srv_msgs/pdu_cdr_conv_JoinRequest.py
from .pdu_pytype_JoinRequest import JoinRequest
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_JoinRequest(writer: CdrWriter, src: JoinRequest):
    writer.write_string(src.name)


def cdr_body_to_py_JoinRequest(reader: CdrReader, dst: JoinRequest):
    dst.name = reader.read_string()
    return dst


def py_to_cdr_JoinRequest(src: JoinRequest) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_JoinRequest(writer, src)
    return writer.bytes()


def cdr_to_py_JoinRequest(cdr_payload) -> JoinRequest:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = JoinRequest()
    return cdr_body_to_py_JoinRequest(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_JoinRequest.js
import { JoinRequest } from './pdu_jstype_JoinRequest.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduJoinRequestConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { JoinRequest } src
     */
    static to_cdr_body(writer, src) {
        writer.write_string(src.name);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { JoinRequest } dst
     * @returns { JoinRequest }
     */
    static cdr_body_to_js(reader, dst) {
        dst.name = reader.read_string();
        return dst;
    }

    /**
     * @param { JoinRequest } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { JoinRequest }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new JoinRequest());
    }
}
